tests/unit/amo/components/TestAddonRecommendations.js

```diff
@@ -0,0 +1,224 @@
+import * as React from 'react';
+
+import AddonRecommendations, {
+  AddonRecommendationsBase,
+  TAAR_IMPRESSION_CATEGORY,
+  TAAR_COHORT_COOKIE_NAME,
+  TAAR_COHORT_INCLUDED,
+  TAAR_COHORT_EXCLUDED,
+} from 'amo/components/AddonRecommendations';
+import {
+  fetchRecommendations,
+  loadRecommendations,
+} from 'amo/reducers/recommendations';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeRecommendations,
+} from 'tests/unit/amo/helpers';
+import {
+  createStubErrorHandler,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+const fakeCookie = (returnValue) => {
+  return {
+    load: sinon.stub().returns(returnValue),
+    save: sinon.spy(),
+  };
+};
+
+const fakeRandomizer = (expectedRecommendedValue) => {
+  return sinon.stub().returns(expectedRecommendedValue ? 0.51 : 0.5);
+};
+
+describe(__filename, () => {
+  let fakeTracking;
+  let store;
+
+  beforeEach(() => {
+    fakeTracking = {
+      sendEvent: sinon.spy(),
+    };
+    store = dispatchClientMetadata().store;
+  });
+
+  function doFetchRecommendations({
```

Comment:
Ah I think that's where the `_` naming is useful, so that it does not conflict with imported functions. But I am fine with `do` as a prefix 😉